### PR TITLE
pump spark version from 3.2.2 to 3.3.1

### PR DESCRIPTION
**Pull request by JessicaXYWang**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

_Briefly describe the changes included in this Pull Request._

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

File: core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksGPUTests.scala [hunk 1]
```diff
@@ -11,11 +11,7 @@ import java.io.File
 import scala.collection.mutable.ListBuffer
 
 class DatabricksGPUTests extends DatabricksTestHelper {
-  val horovodInstallationScript: File = FileUtilities.join(
-    BuildInfo.baseDirectory.getParent, "deep-learning",
-    "src", "main", "python", "horovod_installation.sh").getCanonicalFile
-  uploadFileToDBFS(horovodInstallationScript, "/FileStore/horovod-fix-commit/horovod_installation.sh")
-  val clusterId: String = createClusterInPool(GPUClusterName, AdbGpuRuntime, 2, GpuPoolId, GPUInitScripts)
+  val clusterId: String = createClusterInPool(GPUClusterName, AdbGpuRuntime, 2, GpuPoolId, "[]")
```
Comment (mhamilton723) [inline core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksGPUTests.scala:14]: Was this necessary, it looks like this will remove functionality

File: environment.yml [hunk 2]
```diff
@@ -12,8 +12,7 @@ dependencies:
   - pip:
     - pyarrow>=0.15.0
     - numpy>=1.19.3
-    - pyspark==3.2.3
-    - pandas==1.2.5
```
Comment (mhamilton723) [inline environment.yml:16]: Do we want to remove pandas here?

File: project/Secrets.scala [hunk 3]
```diff
@@ -14,7 +14,7 @@ object Secrets {
   private val KvName = "mmlspark-keys"
   private val SubscriptionID = "e342c2c0-f844-4b18-9208-52c8c234c30e"
   private val PgpFileExtension = ".asc"
-  private val EnablePublishEnvVar = "SYNAPSEML_ENABLE_PUBLISH"
+  private val EnablePublishEnvVar = "SYNAPSEML-ENABLE-PUBLISH"
```
Comment (mhamilton723) [inline project/Secrets.scala:17]: Any idea why this changes? Bash variables are more common with _

**Comment by rabbyn** [comment]
hello @mhamilton723 , my client works with Synapse Spark Clusters version 3.3 and by the doc [here](https://github.com/microsoft/SynapseML/blob/master/README.md#synapse-analytics) I'm not sure if we have to add this cell or not since it's only written for Spark3.2 pool. Thanks

**Comment by vishalovercome** [comment]
My company wants to upgrade our Spark cluster and I wanted to know by when this PR will be merged and released. What are the timelines for spark 3.4?
